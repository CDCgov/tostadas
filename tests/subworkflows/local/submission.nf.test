nextflow_workflow {

    name "Test Workflow SUBMISSION"
    script "subworkflows/local/submission.nf"
    workflow "SUBMISSION"

    test("Run with profile test, virus | biosample") {

        when {
            params {
                submission_prod_or_test = "test"
                send_submission_email = false
                update_submission = false
                fetch_reports_only = false
                biosample = true
                sra = false
                genbank = false
                species = "virus"
                submission_mode = "ftp"
                metadata_basename = "mpxv_test_metadata"
                overwrite_output = true
                dry_run = true
            }
            workflow {
                """
                input[0] = Channel.of(
                            tuple(    
                                [batch_id: 'batch_1', batch_tsv: file("${projectDir}/tests/subworkflows/local/batch_1_mpxv.tsv")], 
                                [ 
                                    [meta: [ sample_id: 'NY0006' ], fq1: file("${projectDir}/assets/sample_fastqs/mpox/2022-028-7666_S3_L001_R1_001.fastq.gz"), fq2: file("${projectDir}/assets/sample_fastqs/mpox/2022-028-7666_S3_L001_R2_001.fastq.gz"), fasta: file("${projectDir}/assets/sample_fastas/mpox/NY0006.fasta"), nnp: null, gff: null], 
                                    [meta: [ sample_id: 'IL0005' ], fq1: file("${projectDir}/assets/sample_fastqs/mpox/LIY15306A2_2022_054_3005007722.R_1.mpx.fastq.gz"), fq2: file("${projectDir}/assets/sample_fastqs/mpox/LIY15306A2_2022_054_3005007722.R_2.mpx.fastq.gz"), fasta: file("${projectDir}/assets/sample_fastas/mpox/IL0005.fasta"), nnp: null, gff: null]
                                ], 
                                ["biosample"]
                            )
                        )
                input[1] = file("${projectDir}/conf/submission_config.yaml")

                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
