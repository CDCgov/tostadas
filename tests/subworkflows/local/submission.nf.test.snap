{
    "Run with profile test, virus | biosample": {
        "content": [
            {
                "0": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1_mpxv.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c"
                        },
                        [
                            [
                                "submission.xml:md5,71d3fcc17f36692d2d1dc276c69313c1",
                                "submit.ready:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ],
                            "prep_submission.log:md5,46670308a40f814e8586ff8e1231018e",
                            "submission.log:md5,e913a908f2bbeea2e2d895679701d9f4"
                        ]
                    ]
                ],
                "submission_batch_folder": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1_mpxv.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c"
                        },
                        [
                            [
                                "submission.xml:md5,71d3fcc17f36692d2d1dc276c69313c1",
                                "submit.ready:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ],
                            "prep_submission.log:md5,46670308a40f814e8586ff8e1231018e",
                            "submission.log:md5,e913a908f2bbeea2e2d895679701d9f4"
                        ]
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-28T16:33:08.667586447"
    }
}